src/arrayMethodJoin.js

```diff
@@ -5,7 +5,16 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sep = separator === undefined
+      ? ',' : separator === null ? 'null' : separator.toString();
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      str = str + sep + (this[i] === null || this[i] === undefined
+        ? '' : this[i]);
+    }
+
+    return str.substring(sep.length, str.length);
```

Comment:
I would create a variable then return it, it would be more readable.

Reply:
Also, you don't have to specify  the second parameter of `.str.substring` (end index) since default means last index of a string already

Reply:
done